supervisely/io/fs.py

```diff
@@ -622,28 +623,34 @@ def archive_directory(
         while True:
             part_name = f"{tar_name}.{str(part_number).zfill(3)}"
             output_path = os.path.join(tar_dir, part_name)
-            data = input_file.read(split)
-            if not data:
-                break
             with open(output_path, "wb") as output_file:
-                output_file.write(data)
+                while output_file.tell() < split:
+                    data = input_file.read(chunk)
+                    if not data:
+                        break
+                    output_file.write(data)
                 parts_paths.append(output_path)
                 part_number += 1
+            if not data:
+                break
 
     os.remove(tar_path)
     return parts_paths
 
 
-def unpack_archive(archive_path: str, target_dir: str, remove_junk=True) -> None:
+def unpack_archive(archive_path: str, target_dir: str, remove_junk=True, is_split=False) -> None:
```

Comment:
split_size_mb = None / or 50 or ....
default 500?

Reply:
Add `chunk_size_mb` parameter to `archive_directory` and `unpack_archive` functions.